helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,176 @@
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+
+    private val targetFormat by option(
+        "--target", "-t",
+        help = "The target format for the conversion. Allowed values: GRAPH."
+    ).enum<TargetFormat>().default(TargetFormat.GRAPH)
+
+    override fun run() {
+        val converters = listOf(this::convertOrtResult, this::convertAnalyzerResult, this::convertProjectAnalyzerResult)
+        if (converters.none { it() }) {
+            failure("No supported data type could be found in the input file.", 1)
+        }
+    }
+
+    /**
+     * Convert a full ORT result file.
+     */
+    private fun convertOrtResult(): Boolean =
+        convertResult<OrtResult> { result ->
+            println("Converting a full ORT result file.")
+
+            val analyzerResult = convertToTarget(result.analyzer?.result)
+            val convertedResult = result.copy(
+                analyzer = result.analyzer?.copy(result = analyzerResult)
+            )
+
+            writeResult(convertedResult)
+        }
+
+    /**
+     * Convert an [AnalyzerResult] file.
+     */
+    private fun convertAnalyzerResult(): Boolean =
+        convertResult<AnalyzerResult> {
+            println("Converting an analyzer result file.")
+
+            writeResult(convertToTarget(it))
+        }
+
+    /**
+     * Convert a [ProjectAnalyzerResult]. In this case, the result file contains an [AnalyzerResult], as a
+     * [ProjectAnalyzerResult] does not contain a dependency graph.
+     */
+    private fun convertProjectAnalyzerResult(): Boolean =
+        convertResult<ProjectAnalyzerResult> { result ->
+            println("Converting a project analyzer result file.")
+
+            val analyzerResult = AnalyzerResultBuilder().addResult(result).build()
+            writeResult(convertToTarget(analyzerResult))
+        }
+
+    /**
+     * Convert a result of type [T]. Read this result from the input file and invoke [block] with it. Return *false*
+     * if the result file could not be loaded (which probably means that it contains a result of a different type).
+     */
+    private inline fun <reified T : Any> convertResult(block: (T) -> Unit): Boolean =
+        runCatching {
+            block(ortFile.readValue())
+        }.fold(onSuccess = { true }, onFailure = { false })
+
+    /**
+     * Convert the given [analyzerResult] to the selected target format. Fail if no result is available or the
+     * result is already in the target format.
+     */
+    private fun convertToTarget(analyzerResult: AnalyzerResult?): AnalyzerResult {
+        val convertedResult = targetFormat.convert(
+            analyzerResult ?: failure("No analyzer result available.", 1)
+        )
+
+        if (convertedResult == analyzerResult) {
+            failure("The result already uses the dependency graph format. No conversion is done.", 2)
+        }
```

Comment:
In [this comment](https://github.com/oss-review-toolkit/ort/pull/4191#discussion_r655095130) you argued that a user should be able to run the comment to ensure a specific output format, independent of the input format. With that in mind, it IMO should be no failure if the input already is in the desired format. Also, the user might have specified a different output file, effective asking for a copy of the input in that case.

Reply:
I wanted the command to return a different exit code when no conversion was done. Currently, exit code 1 is used for real errors and 2 if no conversion was done. This is a bit similar to the handling of issues and hints in ORT's main commands.

Is it desired to have such different exit codes, and - if yes -, what would be the best way to achieve it?

Reply:
IMO it would not be *required* to have a different exit code in this case, but I can see how it makes sense. As I was mostly disturbed by the semantics of the word "failure", I guess my issue would go away if we'd just throw a `UsageError` directly instead.

Reply:
The function has been replaced by throwing `UsageError` directly.